frontend/src/app/assignments/[id]/layout.tsx

```diff
@@ -16,11 +16,11 @@ export default function AssignmentPageLayout({
 export const generateMetadata = async ({
   params,
 }: PageProps): Promise<Metadata> => {
-  const assignmentTitle = (
-    await AssignmentService.getAssignmentById({
-      assignmentId: params.id,
-    })
-  ).title;
+  const currentAssignment = await AssignmentService.getAssignmentById(
+    params.id
+  );
+
+  const assignmentTitle = currentAssignment?.title ?? "No such assignment";
```

Comment:
Are we expecting the page to 404 atm if an assignment is not found?
I see that you made some logic to handle if the assignment is not found here.

Reply:
Yes, I don't think we need to focus on the error page for now.
For this, I am just controlling the page title shown in the browser tab.